blades/monopodws_imxrt.h

```diff
@@ -0,0 +1,143 @@
+// void init_pinlist(){ 
+//   if (bladePin != -1) pinList[numPins++] = bladePin;
+//   if (blade2Pin != -1) pinList[numPins++] = blade2Pin;
+//   if (blade3Pin != -1) pinList[numPins++] = blade3Pin;
+//   if (blade4Pin != -1) pinList[numPins++] = blade4Pin;
+//   if (blade5Pin != -1) pinList[numPins++] = blade5Pin;
+//   if (blade6Pin != -1) pinList[numPins++] = blade6Pin;
+// }
+
+// maxLedPerStrip determines the fps acheivable, do not make it larger than the largest blade
+// ToDo;
+// auto generate maxLedsPerStrip from config file largest blade??
+DMAMEM int displayMemory[maxLedsPerStrip * numPins * 3 / 4];//always reserve for  numpins*3 for RGB, /4 4bytes per int?
+int drawingMemory[maxLedsPerStrip * numPins * 3 / 4];
+const int config_ = WS2811_RGB | WS2811_800kHz; //RGB, ProffieOS already swaps bytes correctly
+OctoWS2811 leds(maxLedsPerStrip, displayMemory, drawingMemory, config_, numPins, pinList);
+uint32_t frame_sync_time_ = 0;//sync to first blade
+uint32_t min_update_time_ = 0;//determined by maxLedsPerStrip * 30us/led + 300us idle time, rounded upward
+uint32_t frame_num_ = 0;      //frame number shared between blades
+
+
+template<Color8::Byteorder BYTEORDER>
+class MonopodWSPinBase : public WS2811PIN {
+public:
+MonopodWSPinBase(int num_leds, int pin, uint32_t frequency,
+           uint32_t reset_us, uint32_t t0h_us, uint32_t t1h_us) :
+    pin_(pin),  //which pin
+    num_leds_(num_leds), //amount of leds for this strip
+    frequency_(frequency) { //800000
+
+    for (int j = 0; j < numPins; j++) {
+        if (pin == pinList[j]) {
+            offset_ = maxLedsPerStrip * j;
+            STDOUT.print("pin: ");
+            STDOUT.print(pin);
+            STDOUT.print(", offset: ");
+            STDOUT.print(offset_);
+            if (j == 0) {
+                is_first_blade_ = true;
+                STDOUT.print(", is first blade");
+                //determined by maxLedsPerStrip * 30us/led + 300us idle time, rounded upward
+                STDOUT.print(", min update time us: ");
+                STDOUT.print((maxLedsPerStrip * 24000000.0 / frequency_ + 300));
+                STDOUT.print(", min update ms: ");
+                //round up to nearest ms interval
+                min_update_time_ = ((maxLedsPerStrip * 24000000.0 / frequency_ + 300)/1000)+1;
+                STDOUT.print(min_update_time_);
+            }
+            if (j == numPins-1) {
+                is_last_blade_ = true;
+                STDOUT.print(", is last blade");
+            }
+            STDOUT.println();
+        }
+    }
+  }
+  
+  bool IsReadyForBeginFrame() {
+        if (is_first_blade_) {
+            return !leds.busy();//sync to 1st blade
+        }else{
+            return true; // other blades can start filling displayMemory
+        }           
+  }
+  
+  bool IsReadyForEndFrame() {
+        return millis() > (frame_sync_time_ + min_update_time_);//4ms =250fps, 5ms=200fps, 
```

Comment:
it would probably be better to do this in micros.

Reply:
I have done a lot of experimenting with micros() vs millis(). 

When using micros() I sometimes get strange behavior of the secondary blades. They instantly extend/retract, while the main blade is working as it should. When syncing to full milliseconds rounded upward, I get the most reliable performance of the system.
I also think the performance is good enough even for higher led-counts like the new KR-ledstrips with 264 pixels.
144 pixels -> 4.62ms update (rounded 5ms), 216 fps theoretical vs 200 fps when rounded up to nearest millisec()
264 pixels -> 8.22ms update (rounded 9ms) 121 fps theoretical vs 111 fps when rounded up to nearest millisec()
This is robustness over performance from my point of view, based on empirical evidence...

Reply:
The problem might have to do with wraparounds.
This code here for instance, won't when millis() is more than 2^32 - 4
Generally, I find that the only safe way to deal with micros() and millis() is to remember a timestamp, and when you wish to know how long ago that timestamp was, you do micros() - timestamp.  So in this case, that becomes:

  millis() - frame_sync_time_ > min_update_time_

As long as frame_sync_time_ is guaranteed to be in the past (and not TOO long in the past), this always works.

Reply:
Yeah, should have done that myself. The way you suggest, the millis() can just roll over but the time difference from millis() - frame_sync_time_   would still be correct because of uint subtraction. 
I will adjust the code to do it like this.